logback-classic/src/main/java/ch/qos/logback/classic/helpers/MDCInsertingServletRequestListener.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Logback: the reliable, generic, fast and flexible logging framework.
+ * Copyright (C) 1999-2015, QOS.ch. All rights reserved.
+ * Copyright (C) 2024 Sven Strickroth <email@cs-ware.de>
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation
+ *
+ *   or (per the licensee's choosing)
+ *
+ * under the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation.
+ */
+package ch.qos.logback.classic.helpers;
+
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletRequestEvent;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.http.HttpServletRequest;
+
+import org.slf4j.MDC;
+
+import ch.qos.logback.classic.ClassicConstants;
+
+/**
+ * A simple servlet request listener that stores the remote address,
+ * used HTTP method, and 
+ *
+ * <p> The value is removed from the MDC once the request has
+ * been fully processed (including an error handler servlet).
+ *
+ * @author Sven Strickroth
+ * @author Ceki G&uuml;lc&uuml;
+ */
+public class MDCInsertingServletRequestListener implements ServletRequestListener {
+    @Override
+    public void requestInitialized(ServletRequestEvent sre) {
+        insertIntoMDC(sre.getServletRequest());
```

Comment:
Here, we could also cann the `insertIntoMDC` in the `MDCInsertingServletFilter` to avoid code duplication.